cmd/quill/internal/ui/ui.go

```diff
@@ -86,11 +87,26 @@ func (m *UI) Teardown(force bool) error {
 	if !force {
 		m.handler.State().Running.Wait()
 		m.program.Quit()
+		// typically in all cases we would want to wait for the UI to finish. However there are still error cases
+		// that are not accounted for, resulting in hangs. For now, we'll just wait for the UI to finish in the
+		// happy path only. There will always be an indication of the problem to the user via reporting the error
+		// string from the worker (outside of the UI after teardown).
+		m.running.Wait()
 	} else {
-		m.program.Kill()
-	}
+		_ = runWithTimeout(250*time.Millisecond, func() error {
```

Comment:
Just to make sure I understand the change, we're basically saying, "if we're exiting in a controlled manner, wait for the UI to finish. Otherwise, wait 250 milliseconds for the UI to finish," so that the UI has a better chance to restore the terminal to its normal state?

Reply:
Yes. If we're in a "force shutdown" mode, wait only a short time for the handler and UI to complete. This short time, however, is ages longer than the near instantaneous end of the main thread in the program, which was the culprit for leaving the terminal in a bad state.

Reply:
Thanks for the explanation!